### PR TITLE
Initial documentation for list builder and related classes

### PR DESCRIPTION
**Pull request by msljivic**

### CODE REVIEW

**Review by msljivic — COMMENTED** [review]
Add class level documentation and missing documentation for relevant methods.

File: Enigmatry.Entry.CodeGeneration.Configuration/Builder/BaseComponentBuilder.cs [hunk 1]
```diff
@@ -13,6 +13,10 @@ protected BaseComponentBuilder(Type modelType)
         _componentInfoBuilder = new ComponentInfoBuilder(modelType);
     }
 
+    /// <summary>
+    /// Configure component-level options, such as the component name, feature name, and translation ID.
```
Comment (markovic-katarina) [inline Enigmatry.Entry.CodeGeneration.Configuration/Builder/BaseComponentBuilder.cs:17]: Add class level documentation and missing documentation for relevant methods.

File: Enigmatry.Entry.CodeGeneration.Configuration/Builder/ComponentInfoBuilder.cs
Comment (markovic-katarina) [inline Enigmatry.Entry.CodeGeneration.Configuration/Builder/ComponentInfoBuilder.cs:1]: Perhaps add default values and possible Exceptions.

File: Enigmatry.Entry.CodeGeneration.Configuration/Formatters/CurrencyPropertyFormatter.cs [hunk 2]
```diff
@@ -3,14 +3,38 @@
 
 namespace Enigmatry.Entry.CodeGeneration.Configuration.Formatters;
 
+/// <summary>
+/// The CurrencyPropertyFormatter formats numeric values as currency values in a table column.
```
Comment (markovic-katarina) [inline Enigmatry.Entry.CodeGeneration.Configuration/Formatters/CurrencyPropertyFormatter.cs:7]: Why just in a table column? Check where it's used.

File: Enigmatry.Entry.CodeGeneration.Configuration/Formatters/CurrencyPropertyFormatter.cs
Comment (markovic-katarina) [inline Enigmatry.Entry.CodeGeneration.Configuration/Formatters/CurrencyPropertyFormatter.cs:1]: Add example/s for class properties.

File: Enigmatry.Entry.CodeGeneration.Configuration/Formatters/DatePropertyFormatter.cs
Comment (markovic-katarina) [inline Enigmatry.Entry.CodeGeneration.Configuration/Formatters/DatePropertyFormatter.cs:1]: Add examples for class properties.

File: Enigmatry.Entry.CodeGeneration.Configuration/Formatters/IPropertyFormatter.cs
Comment (markovic-katarina) [inline Enigmatry.Entry.CodeGeneration.Configuration/Formatters/IPropertyFormatter.cs:1]: Is it really used just in the table component or can they be used in other components, e.g. Form component.

File: Enigmatry.Entry.CodeGeneration.Configuration/Formatters/PercentPropertyFormatter.cs
Comment (markovic-katarina) [inline Enigmatry.Entry.CodeGeneration.Configuration/Formatters/PercentPropertyFormatter.cs:1]: Add examples.

File: Enigmatry.Entry.CodeGeneration.Configuration/Formatters/IPropertyFormatter.cs
Comment (markovic-katarina) [reply Enigmatry.Entry.CodeGeneration.Configuration/Formatters/IPropertyFormatter.cs:1]: It is used both in formatting the Table columns as well as Form controls.

**Review by markovic-katarina — APPROVED** [review]
Approve after pear review with Andries and Miljan.

**Pull request merged**